doc/esmvalcore/preprocessor.rst

```diff
@@ -138,344 +260,559 @@ losing generality. To mask ice out, `mask_landseaice` can be used:
-seasonal_mean function, below.
+If you want the seasonal average, then this function needs to be combined with
+the seasonal_mean function, below.
 
 See also :func:`esmvalcore.preprocessor.extract_season`.
 
-3. extract_month
-----------------
+``extract_month``
+-----------------
 
 The function extracts the times that occur within a specific month.
-This function only has one argument:
-
-* month: [1-12]
-
-Note that named month strings will not be accepted.
+This function only has one argument: ``month``. This value should be an integer
+between 1 and 12 as the named month string will not be accepted.
 
 See also :func:`esmvalcore.preprocessor.extract_month`.
 
-4. time_average
----------------
+.. _time_average:
+
+``time_average``
+----------------
 
-This function takes the weighted average over the time dimension. This
+This functions takes the weighted average over the time dimension. This
 function requires no arguments and removes the time dimension of the cube.
 
 See also :func:`esmvalcore.preprocessor.time_average`.
 
-5. seasonal_mean
-----------------
+``seasonal_mean``
+-----------------
 
 This function produces a seasonal mean for each season (DJF, MAM, JJA, SON).
 Note that this function will not check for missing time points. For instance,
-if the DJF field is selected, but the input datasets starts on January 1st,
+if you are looking at the DJF field, but your datasets starts on January 1st,
 the first DJF field will only contain data from January and February.
 
 We recommend using the extract_time to start the dataset from the following
 December and remove such biased initial datapoints.
 
 See also :func:`esmvalcore.preprocessor.seasonal_mean`.
 
-6. annual_mean
---------------
+``annual_mean``
+---------------
 
-This function produces an annual or a decadal mean. It takes a single boolean
-switch as argument:
-* decadal: set this to true to calculate decadal averages instead of annual
-averages.
+This function produces an annual or a decadal mean. The only argument is the
+decadal boolean switch. When this switch is set to true, this function
+will output the decadal averages.
 
 See also :func:`esmvalcore.preprocessor.annual_mean`.
 
-7. regrid_time
---------------
-
-This function aligns the time points of each component dataset to allow the
-subtraction of two Iris cubes from different datasets. The operation makes the
-datasets time points common and sets common calendars; it also resets the time
-bounds and auxiliary coordinates to reflect the artificially shifted time
-points. The current implementation works for monthly, daily, 6 hourly, 3
-hourly and hourly data. It takes a string representing the data frequency as
-an input argument:
-* frequency: mon, day, 1hr, 3hr, or 6hr 
-        
+``regrid_time``
+---------------
+
+This function aligns the time points of each component dataset so that the
+dataset Iris cubes can be subtracted. The operation makes the datasets time
```

Comment:
Sounds weird "... so that the dataset Iris cubes can be subtracted." Does it mean that the Iris cubes from different datasets can be subtracted?

Reply:
Rephrased.